### DISCUSSION TITLE
LE Script ignores Lidarr's blocklist

### DISCUSSION THREAD

**Discussion by Yosh82** [RandomNinjaAtk/docker-lidarr-extended#138] [Ideas]
Hi. Is there a way to stop artists from being re-added, that have already been added to the Lidarr exclusion list?

Example: 
Artist1 is added to Lidarr by the LE script as part of the top 20 artists from Deezer.
I delete Artist1 from Lidarr and add to Lidarr exclusion list. Files deleted and artists removed from Lidarr.
When it next scan's, it adds Artist1 again to Lidarr and downloads the files again, as it's still in the Top20 on Deezer, and it's still in Lidarr's blocklist.

I didn't want to raise it as bug, as it makes a bit of sense to why thinking it through, but would be nice to have a solution if possible.

Thanks in advance.... and still loving LE!

**Comment by RandomNinjaAtk** [comment]
Hmm, that is a good idea, I'd have to look into seeing if its possible to check against the block list to prevent it from happening...

**Comment by RandomNinjaAtk** [comment]
[v1.0.005](https://github.com/RandomNinjaAtk/docker-lidarr-extended/commit/fe546ab0e850fbbe23f22edd530d74a9933c3ace)

The v1.0.005 script update should accomplish your feature request, you'll have to test it though....

**Reply by Yosh82** [reply]
You sir, are a gent. I will test and update this thread. Many thanks!

**Reply by Yosh82** [reply]
![image](https://user-images.githubusercontent.com/106485860/203646510-2d305d28-a509-45b6-bbc3-2822b6051438.png)
Amazing. I'll let it complete the scan and report if any issues, but looks good (I'm on 2.5k artists thanks to you / LE, so will report back tomorrow as will take a while).

Thanks again.

**Reply by Yosh82** [reply]
Seems great, no issues on testing. 
One thing I haven't tested, which I didn't request but may already be part of the update, is the same for any albums/singles etc that are in the blocklist (some artists have foreign language albums which I delete and block, and the awful xmas albums too 😄 ).

**Reply by RandomNinjaAtk** [reply]
Unmonitoring, instead of blocking will solve that. The script only looks for wanted and cutoff items.

**Reply by Yosh82** [reply]
Ok thanks. Previously I was un-monitoring them, but as it downloaded the files already, I then deleted from Lidarr and blocked at the same time. I can manually delete once un-monitored, not an issue.

**Reply by Yosh82** [reply]
Just to report, all working perfectly with this change since testing for nearly 24 hours. 👍
